conn.go

```diff
@@ -248,14 +281,22 @@ func (c *conn) buildRequest(ctx context.Context, query string, params []driver.V
 	}
 	c.log("query: ", query)
 	req, err := http.NewRequest(method, c.url.String(), strings.NewReader(query))
+	if err != nil {
+		return nil, err
+	}
 	// http.Transport ignores url.User argument, handle it here
-	if err == nil && c.user != nil {
+	if c.user != nil {
 		p, _ := c.user.Password()
 		req.SetBasicAuth(c.user.Username(), p)
 	}
+	setQueryIDCond := c.killQueryOnTimeout && c.transport.ResponseHeaderTimeout > 0
 	if ctx != nil {
 		quotaKey, quotaOk := ctx.Value(QuotaKey).(string)
```

Comment:
this block is getting pretty messy. What do you think about refactoring it to something like this 
```
var queryID, quotaKey string
if ctx != nil {
    quotaKey, _ = ctx.Value(QuotaKey).(string)
	queryID, _ = ctx.Value(QueryID).(string)
}

if c.killQueryOnTimeout && queryID == "" {
    queryID := uuid.NewV4().String()
}

reqQuery := req.URL.Query()
if quotaKey != "" {
    reqQuery.Add(quotaKeyParamName, quotaKey)
}
if queryID != "" {
    reqQuery.Add(queryIDParamName, queryID)
}
req.URL.RawQuery = reqQuery.Encode()

return req, nil
```